src/pty.rs

```diff
@@ -0,0 +1,50 @@
+use libc;
+
+use {Errno, Result};
+use std::os::unix::io::RawFd;
+
+use sys::termios::Termios;
+
+pub use libc::pid_t as SessionId;
+pub use libc::winsize as Winsize;
+
+pub struct OpenptyResult {
+    pub master: RawFd,
+    pub slave: RawFd,
+}
+
+/// Create a new pseudoterminal, returning the slave and master file descriptors
+/// in `OpenptyResult`
+/// (see [openpty](http://man7.org/linux/man-pages/man3/openpty.3.html)). 
+///
+/// If `winsize` is not `None`, the window size of the slave will be set to
+/// the values in `winsize`. If `termios` is not `None`, the pseudoterminal's
+/// terminal settings of the slave will be set to the values in `termios`.
+#[inline]
+pub fn openpty(winsize: Option<Winsize>, termios: Option<Termios>) -> Result<OpenptyResult> {
```

Comment:
First, can't both of be borrows?

Second, we should be able to use a slightly different function declaration to give us better ergonomics when using a real struct for either of the arguments, something like:
```
pub fn openpty<T: Into<Option<&Winsize>>, U: Into<Option<&Termios>>>(winsize: T, termios: U) -> ... {
```
So now you can use `.into()` to convert the options to the struct representations and the caller doesn't need to wrap their struct reference in a `Some`. I don't know if the syntax above is exactly correct, but I think it should be. The [gtk-rs](https://github.com/gtk-rs/gtk) project did this quite successfully for all their string types and it's nice ([here](https://github.com/gtk-rs/gtk/blob/a40a4742ba09e11380e4cb1c51628e4aa4851d19/src/auto/tool_button.rs#L28) is an example).